examples/nemo/scripts/process_nemo.py

```diff
@@ -0,0 +1,147 @@
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+
+'''
+Python driver script to run PSyclone on (pre-processed) NEMO source files, i.e.
+those produced by the build system in <MY_CONFIG_NAME>/BLD/ppsrc/nemo.
+For example::
+
+>>> process_nemo.py -s ./kernels_trans.py -o <MY_CONFIG_NAME>/MY_SRC \
+  <MY_CONFIG_NAME>/BLD/ppsrc/nemo/*90
+
+Or, if you have Gnu 'parallel':
+
+>>> parallel process_nemo.py -s ./kernels_trans.py -o <MY_CONFIG_NAME>/MY_SRC \
+  {} ::: <MY_CONFIG_NAME>/BLD/ppsrc/nemo/*90
+
+'''
+
+from __future__ import print_function
+import os
+import sys
+
+# Whether or not we attempt to add profiling to *every* file (i.e. including
+# those listed under EXCLUDED_FILES below).
+PROFILE_ALL = True
+
+# Files that we won't attempt to process with PSyclone
+EXCLUDED_FILES = ["bdyini.f90",
+                  "diaobs.f90",
+                  "diawri.f90",  # Unused & has Return in profile region
+                  "dommsk.f90",
+                  "fldread.f90",
+                  "icbclv.f90", "icbdyn.f90", "icblbc.f90", "icbrst.f90",
+                  "icbthm.f90", "icbutl.f90", "icbdia.f90", "icbini.f90",
+                  "icb_oce.f90", "icbstp.f90", "icbtrj.f90",
+                  "ice.f90",  # lines are too long
+                  "icedyn_adv_pra.f90",  # lines are too long
+                  "iceforcing.f90",
+                  "iceistate.f90",
+                  "icethd_ent.f90", "icethd_zdf.f90",
+                  "icethd_dh.f90", "iom.f90",
+                  "mppini.f90",  # MPI code
+                  "obs_inter_h2d.f90", "obs_grid.f90", "obs_averg_h2d.f90",
+                  "obs_profiles_def.f90", "obs_sort.f90", "obs_types.f90",
+                  "obs_utils.f90",
+                  "storng.f90", "stopar.f90",
+                  "tide_mod.f90", "timing.f90"]
+
+if __name__ == "__main__":
+    import argparse
+
+    PARSER = argparse.ArgumentParser(
+        description="Process the specified NEMO source files "
+        "using PSyclone")
+    PARSER.add_argument('input_files', metavar='input_file', type=str,
+                        nargs='+',
+                        help="One or more NEMO pre-processed source files")
+    PARSER.add_argument('-o', dest='out_dir',
+                        help="Destination directory for processed source "
+                        "files")
+    PARSER.add_argument('-s', dest='script_file',
+                        help="PSyclone transformation script")
+    PARSER.add_argument('-x', dest='exit_on_error', action='store_true',
+                        help="eXit immediately if PSyclone fails")
```

Comment:
As mentioned before, you could replace the X with x

Reply:
Done.